deployment/hasura/migrations/AerieMerlin/23_activity_directive_changelog/down.sql

```diff
@@ -0,0 +1,1006 @@
+  using merge_staging_area msa
+  where ad.id = msa.activity_id
+    and ad.plan_id = plan_id_R
+    and msa.merge_request_id = _request_id
+    and msa.change_type = 'delete';
+
+  -- Clean up
+  delete from conflicting_activities where merge_request_id = _request_id;
+  delete from merge_staging_area where merge_staging_area.merge_request_id = _request_id;
+
+  update merge_request
+  set status = 'accepted'
+  where id = _request_id;
+
+  -- Attach snapshot history
+  insert into plan_latest_snapshot(plan_id, snapshot_id)
+  select plan_id_receiving_changes, snapshot_id_supplying_changes
+  from merge_request
+  where id = _request_id;
+end
+$$;
+
+create or replace function create_snapshot(_plan_id integer)
+  returns integer -- snapshot id inserted into the table
+  language plpgsql as $$
+  declare
+    validate_plan_id integer;
+    inserted_snapshot_id integer;
+begin
+  select id from plan where plan.id = _plan_id into validate_plan_id;
+  if validate_plan_id is null then
+    raise exception 'Plan % does not exist.', _plan_id;
+  end if;
+
+  insert into plan_snapshot(plan_id, revision, name, duration, start_time)
+    select id, revision, name, duration, start_time
+    from plan where id = _plan_id
+    returning snapshot_id into inserted_snapshot_id;
+  insert into plan_snapshot_activities(
+                snapshot_id, id, name, source_scheduling_goal_id, created_at, last_modified_at, start_offset, type,
+                arguments, last_modified_arguments_at, metadata, anchor_id, anchored_to_start)
+    select
+      inserted_snapshot_id,                              -- this is the snapshot id
+      id, name, source_scheduling_goal_id, created_at,   -- these are the rest of the data for an activity row
+      last_modified_at, start_offset, type, arguments, last_modified_arguments_at, metadata, anchor_id, anchored_to_start
+    from activity_directive where activity_directive.plan_id = _plan_id;
+  insert into preset_to_snapshot_directive(preset_id, activity_id, snapshot_id)
+    select ptd.preset_id, ptd.activity_id, inserted_snapshot_id
+    from preset_to_directive ptd
+    where ptd.plan_id = _plan_id;
+  insert into metadata.snapshot_activity_tags(snapshot_id, directive_id, tag_id)
+    select inserted_snapshot_id, directive_id, tag_id
+    from metadata.activity_directive_tags adt
+    where adt.plan_id = _plan_id;
+
+  --all snapshots in plan_latest_snapshot for plan plan_id become the parent of the current snapshot
+  insert into plan_snapshot_parent(snapshot_id, parent_snapshot_id)
+    select inserted_snapshot_id, snapshot_id
+    from plan_latest_snapshot where plan_latest_snapshot.plan_id = _plan_id;
+
+  --remove all of those entries from plan_latest_snapshot and add this new snapshot.
+  delete from plan_latest_snapshot where plan_latest_snapshot.plan_id = _plan_id;
+  insert into plan_latest_snapshot(plan_id, snapshot_id) values (_plan_id, inserted_snapshot_id);
+
+  return inserted_snapshot_id;
+  end;
+$$;
```

Comment:
This section of the migration is not inverted compared to the up

Reply:
Reordered most of the `down.sql` file and it _should_ be good now 👍

Reply:
It's perfect now, thanks!